index.html

```diff
@@ -1,1688 +1,2240 @@
+            pressed or released state, the [=user agent=] MUST choose a
+            <dfn>button press threshold</dfn> to indicate the button as pressed
+            when its value is above a certain amount. If the platform API gives
+            a recommended value, the user agent SHOULD use that. In other
+            cases, the user agent SHOULD choose some other reasonable value.
+          </p>
+          <p>
+            The {{GamepadButton/pressed}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{GamepadButton/[[pressed]]}}.
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>touched</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            The touched state of the button. If the button is capable of
+            detecting touch, this property MUST be `true` if the button is
+            currently being touched, and `false` otherwise. If the button is
+            not capable of detecting touch and is capable of reporting an
+            analog value, this property MUST be `true` if the value property is
+            greater than 0, and `false` if the value is 0. If the button is not
+            capable of detecting touch and can only report a digital value,
+            this property MUST mirror the {{GamepadButton/pressed}} attribute.
+          </p>
+          <p>
+            The {{GamepadButton/touched}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{GamepadButton/[[touched]]}}.
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>value</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            For buttons that have an analog sensor, this property MUST
+            represent the amount which the button has been pressed. All button
+            values MUST be linearly normalized to the range [0.0 .. 1.0]. 0.0
+            MUST mean fully unpressed, and 1.0 MUST mean fully pressed. For
+            buttons without an analog sensor, only the values 0.0 and 1.0 for
+            fully unpressed and fully pressed respectively, MUST be provided.
+          </p>
+          <p>
+            The {{GamepadButton/value}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{GamepadButton/[[value]]}}.
+            </li>
+          </ol>
+        </dd>
+      </dl>
+    </section>
+    <section data-dfn-for="GamepadHapticActuator" data-link-for="GamepadHapticActuator">
+      <h2>
+        <dfn>GamepadHapticActuator</dfn> Interface
+      </h2>
+      <p>
+        A {{GamepadHapticActuator}} corresponds to a configuration of motors or
+        other actuators that can apply a force for the purposes of haptic
+        feedback.
+      </p>
+      <pre class="idl">
+        [Exposed=Window]
+        interface GamepadHapticActuator {
+          readonly attribute GamepadHapticActuatorType type;

```

Comment:
Let's remove `GamepadHapticActuatorType type`, it is not needed now that we have `canPlayEffectType`.